app/src/main/java/ch/epfl/sweng/SDP/auth/Account.java

```diff
@@ -0,0 +1,338 @@
+                                    usersRef);
+                            starsBuilder.addChildren(userId + ".stars").build()
+                                    .setValue(newStars, createCompletionListener());
+                            stars = newStars;
+
+                            localDbHandler.saveAccount(instance);
+                        }
+                    }
+
+                    @Override
+                    public void onCancelled(@NonNull DatabaseError databaseError) {
+                        throw databaseError.toException();
+                    }
+                });
+    }
+
+    /**
+     * Method that allows one to add friends.
+     *
+     * @param usernameId String specifying FirebaseUser.UID of friend
+     * @throws DatabaseException in case write to database fails
+     */
+    public void addFriend(final String usernameId)
+            throws IllegalArgumentException, DatabaseException {
+        if (usernameId == null) {
+            throw new IllegalArgumentException();
+        }
+        DatabaseReferenceBuilder builder = new DatabaseReferenceBuilder(usersRef);
+        builder.addChildren(userId + ".friends." + usernameId).build()
+                .setValue(true, createCompletionListener());
+    }
+
+    /**
+     * Method that allows one to remove friends.
+     *
+     * @param usernameId String specifying FirebaseUser.UID of friend
+     * @throws DatabaseException in case write to database fails
+     */
+    public void removeFriend(final String usernameId)
+            throws IllegalArgumentException, DatabaseException {
+        if (usernameId == null) {
+            throw new IllegalArgumentException();
+        }
+        DatabaseReferenceBuilder builder = new DatabaseReferenceBuilder(usersRef);
+        builder.addChildren(userId + ".friends." + usernameId).build()
+                .removeValue(createCompletionListener());
+    }
+
+    /**
+     * Checks if databaseError occurred.
+     *
+     * @param databaseError potenial databaseError
+     * @throws DatabaseException in case databaseError is non-null
+     */
+    private void checkForDatabaseError(@Nullable DatabaseError databaseError)
+            throws DatabaseException {
+        if (databaseError != null) {
+            throw databaseError.toException();
+        }
+    }
+
+    /**
+     * Creates a CompletionListener that checks if there was a DatabaseError.
+     *
+     * @return CompletionListener
+     */
+    private DatabaseReference.CompletionListener createCompletionListener() {
+        return new DatabaseReference.CompletionListener() {
+            @Override
+            public void onComplete(@Nullable DatabaseError databaseError,
+                    @NonNull DatabaseReference databaseReference) {
+                checkForDatabaseError(databaseError);
+            }
+        };
+    }
+
+    /**
+     * Enable testing.
+     */
+    public static void enableTesting() {
```

Comment:
This is really not good, but I see that it is currently necessary. We should definitely consider creating a "sibling" of this class, which can be used for testing.